ops/pebble.py

```diff
@@ -2335,6 +2369,19 @@ def _wait_change_using_polling(
 
         raise TimeoutError(f'timed out waiting for change {change_id} ({timeout} seconds)')
 
+    def _checks_action(self, action: str, checks: Iterable[str]) -> List[str]:
+        if isinstance(checks, (str, bytes)) or not hasattr(checks, '__iter__'):
+            raise TypeError(f'checks must be of type Iterable[str], not {type(checks).__name__}')
```

Comment:
If each check name is a `str`, why do we type-check against `bytes`?
I recall there's a lot of `str/bytes` duality in pebble, though I thought that was merely historic.

Reply:
This was to be consistent with `_services_action` and `send_signal`.

However, you're right - you can get a series of str via `for x in some_str` but you can't get a series of str via `for x in some_bytes`, so it's more useful to error out in the first test than when looking at what would be the first byte.